src/V3Options.cpp

```diff
@@ -1574,6 +1575,13 @@ void V3Options::parseOptsList(FileLine* fl, const string& optdir, int argc, char
         V3Error::pretendError(V3ErrorCode::UNUSEDSIGNAL, false);
         V3Error::pretendError(V3ErrorCode::UNUSEDPARAM, false);
     });
+    DECL_OPTION("-Wwarn-WIDTH", CbCall, []() {
+        FileLine::globalWarnOff(V3ErrorCode::WIDTH, false);
+        V3Error::pretendError(V3ErrorCode::WIDTHTRUNC, false);
+        V3Error::pretendError(V3ErrorCode::WIDTHEXPAND, false);
+        V3Error::pretendError(V3ErrorCode::WIDTHXZEXPAND, false);
+        V3Error::pretendError(V3ErrorCode::WIDTH, false);
```

Comment:
Yeah I started out following https://github.com/verilator/verilator/pull/3607, but I don't think all of the added functions were completely necessary. I think it might be good to clean that up and align these better in a later PR.

Reply:
Probably the clean way would be to make a map of WIDTHNEW -> WIDTH and likewise for every warning remap, rather than using code, so we can generically handle all remapped messages.

Reply:
did you want me to try to do this on this PR? I was thinking it'd be better to do that on a separate one. Other than that I think it's ready to go.